app/Voucher.php

```diff
@@ -81,6 +80,27 @@ function ($code) {
         );
     }
 
+    /**
+     * Splits a voucher code up
+     *
+     * @param string $code
+     * @return array|bool
+     */
+    public static function splitShortcodeNumeric(string $code)
+    {
+        // Clean the code
+        $clean = self::cleanCodes([$code]);
+        $code = array_shift($clean);
+        // Init matches
+        $matches = [];
+        // split into named matche and return
+        if (preg_match("/^(?<shortcode>\D*)(?<number>\d+)$/", $code, $matches) == 1) {
```

Comment:
I did some manual testing on it. It uses fancy named group return values. Coild do with an auto test. Ill write one seperatly